### ISSUE TITLE
Bug: Trends not working

### ISSUE THREAD

**Issue by joethreepwood** [PostHog/posthog#9166]
## Bug description

Trends aren't working. 

If I load it up, I see this: 

<img width="1209" alt="Screenshot 2022-03-22 at 10 12 44" src="https://user-images.githubusercontent.com/84011561/159457792-fee0f2f9-d1c5-4b7d-9358-07c496a0938a.png">

Showing pageview totals over 7 days. Note the number at the bottom of the screen. 

I then add a filter to just show pageviews for https://posthog.com/handbook/strategy/roadmap and I get this:

<img width="1211" alt="Screenshot 2022-03-22 at 10 11 23" src="https://user-images.githubusercontent.com/84011561/159457932-1a7ce589-a216-46a7-9fd2-c71fb32a5bd0.png">

No graph, and the numbers aren't updated. 

If I change the insight to consider something else, like pageviews counted by initial referring domain = http://google.com then it still looks the same as above. 

If I try adding multiple graph series...

<img width="1213" alt="Screenshot 2022-03-22 at 10 12 58" src="https://user-images.githubusercontent.com/84011561/159458145-a4530ff5-52a5-4148-b7a2-b797295f95c1.png">

Still the same. Trends aren't working. 

## How to reproduce

1. As above
2.
3.

## Environment

- [ ] PostHog Cloud
- [ ] self-hosted PostHog (ClickHouse-based), version/commit: _please provide_
- [ ] self-hosted PostHog (Postgres-based, legacy), version/commit: _please provide_

**Comment by Twixes** [comment]
Looks like this is [this error in Sentry](https://sentry.io/organizations/posthog2/issues/3121811178/?query=is%3Aunresolved+google&statsPeriod=14d).
`DB::Exception: Missing columns: 'alias_mat__current_url' while processing query: 'SELECT DISTINCT alias_mat__current_url FROM posthog.sharded_events [...]`
CC @macobo

**Comment by hazzadous** [comment]
@joethreepwood would you mind trying again? We have pushed a fix which should resolve the issue 🙏

**Comment by joethreepwood** [comment]
Fixed!

**Comment by macobo** [comment]
This is now fixed. Copy-pasta from slack:

1. We need to land https://github.com/PostHog/posthog/pull/8855 to remove using $ signs in variable names to support 21.11 which added heredoc support.
1. The initial approach was to RENAME columns - but this resulted in a deadlocked cluster in production which Harry and James are cleaning up. Info here: https://posthog.slack.com/archives/C01MM7VT7MG/p1647938629144409
1. The next idea was to create ALIAS columns instead - clickhouse sees columns without $ signs in them and internally replaces them with old equivelents. This is a pure metadata change, no files on disk get rewritten = safer.
1. However we ran into a ClickHouse bug (repro below) where in 21.6 ALIAS column is not replaced in some parts of the query and not in others. Frustratingly, this has been fixed in 21.11.
1. I changed materialized column metadata around as a quick fix to still be using $-signed materialized columns which stopped the errors for now.

<details><summary>Here's details on reproducing the bug</summary>

Repro:
```sql
CREATE TABLE test_distributed
(
    `event` String,
    `alias_event` String ALIAS `event`
)
ENGINE = Distributed('posthog', 'default', 'sharded_events')


EXPLAIN SYNTAX SELECT alias_event FROM test_distributed WHERE alias_event != ''
```

In 21.11:
```
┌─explain─────────────────────┐
│ SELECT event AS alias_event │
│ FROM test_distributed       │
│ WHERE alias_event != ''     │
└─────────────────────────────┘
```
In 21.6
```
┌─explain───────────────┐
│ SELECT alias_event    │
│ FROM test_distributed │
│ WHERE event != ''     │
└───────────────────────┘
```

</details>
